src/main/java/seedu/address/logic/commands/DeletePairCommand.java

```diff
@@ -0,0 +1,99 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC_VOLUNTEER;
+
+import java.util.Objects;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.pair.Pair;
+import seedu.address.model.person.Elderly;
+import seedu.address.model.person.Volunteer;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+import seedu.address.model.person.information.Nric;
+
+/**
+ * Deletes the pair identified using NRIC of its elderly and volunteer from the FriendlyLink database.
+ */
+public class DeletePairCommand extends Command {
+    public static final String COMMAND_WORD = "delete_pair";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the pair identified by the nric of an elderly and their volunteer.\n"
+            + "Parameters: "
+            + PREFIX_NRIC_ELDERLY + "ELDERLY ID "
+            + PREFIX_NRIC_VOLUNTEER + "VOLUNTEER ID \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NRIC_ELDERLY + "s02133334I "
+            + PREFIX_NRIC_VOLUNTEER + "T2245343a ";
+
+    public static final String MESSAGE_DELETE_PAIR_SUCCESS = "Deleted Pair: %1$s";
+
+    public static final String MESSAGE_ELDERLY_NOT_FOUND =
+            "The elderly with the specified NRIC does not exist in FriendlyLink";
+
+    public static final String MESSAGE_VOLUNTEER_NOT_FOUND =
+            "The volunteer with the specified NRIC does not exist in FriendlyLink";
+
+    public static final String MESSAGE_PAIR_NOT_FOUND =
+            "The pair with the specified elderly and volunteer does not exist in FriendlyLink";
+
+    private final Nric elderlyNric;
+    private final Nric volunteerNric;
+
+    /**
+     * Constructs a DeletePairCommand to delete a pair.
+     *
+     * @param elderlyNric Nric of the elderly of the pair.
+     * @param volunteerNric Nric of the volunteer of the pair.
+     */
+    public DeletePairCommand(Nric elderlyNric, Nric volunteerNric) {
+        requireAllNonNull(elderlyNric, volunteerNric);
+        this.elderlyNric = elderlyNric;
+        this.volunteerNric = volunteerNric;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Elderly elderly;
+        Volunteer volunteer;
+
+        try {
+            elderly = model.getElderly(elderlyNric);
+        } catch (PersonNotFoundException e) {
+            throw new CommandException(MESSAGE_ELDERLY_NOT_FOUND);
+        }
+
+        try {
+            volunteer = model.getVolunteer(volunteerNric);
+        } catch (PersonNotFoundException e) {
+            throw new CommandException(MESSAGE_VOLUNTEER_NOT_FOUND);
+        }
+
+        Pair toDelete = new Pair(elderly, volunteer);
+        if (!model.hasPair(toDelete)) {
+            throw new CommandException(MESSAGE_PAIR_NOT_FOUND);
+        }
+
+        model.deletePair(toDelete);
```

Comment:
Actually, what do you think about putting the logic of validating the pair to be deleted inside `deletePair`? Also, is there a need to create a new `Pair` object or just validate whether the pair contains the specified elderly and volunteer will do?

Reply:
Yeah I feel maybe validating the elderly and volunteer would be better. Currently pair just contains elderly and volunteer information, but later it may contains more information like time slot etc, so it would be much trouble to create a Pair()

Reply:
thanks for the great suggestions! Hav abstracted this functionality out as well as the above code checking for elderly and volunteer

```java
        try {
            model.deletePair(elderlyNric, volunteerNric);
            return new CommandResult(String.format(MESSAGE_DELETE_PAIR_SUCCESS, elderlyNric, volunteerNric));
        } catch (ElderlyNotFoundException e) {
            throw new CommandException(String.format(MESSAGE_ELDERLY_NOT_FOUND, elderlyNric));
        } catch (VolunteerNotFoundException e) {
            throw new CommandException(String.format(MESSAGE_VOLUNTEER_NOT_FOUND, volunteerNric));
        } catch (PairNotFoundException e) {
            throw new CommandException(String.format(MESSAGE_PAIR_NOT_FOUND, elderlyNric, volunteerNric));
        }
```